### PR TITLE
Feature/119-improve-projects-list-output

### LINKED ISSUE #119: [BE] Make the "List of Projects" to return a real output
As product owners
We want the user detail view to have real data from projects list
To do so, we will make projects to query the database and return the information for the queried user keeping the current output structure
This requirement will be met when the return of the project's list can return real data

### PR DESCRIPTION
**Pull request by coxen34**
# Description
This PR was made to resolve #119, this endpoint was returning false tag data and now all the data it returns is real.

## The changes made are:
* Updated the controller `StudentProjectsDetailController` leaving only the error catch.
* Moved logic to service layer `app\service\Resume\StudentProjectsDetailService`.
* Updated documentation in swagger. `AnnotationsStudentProjectsDetail`.
* Updated  `Feature\Student\StudentProjectsDetailControllerTest` added `unit\Service\StudentProjectsDetailServiceTest` and updated `Fixtures\Resumes`.

### CODE REVIEW

File: app/Exceptions/ProjectNotFoundException.php [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<?php
+
```
Comment (toniGitH) [inline app/Exceptions/ProjectNotFoundException.php:2]: declare(strict_types=1);

File: app/Exceptions/ProjectNotFoundException.php [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+namespace App\Exceptions;
+
+use Exception;
+use Throwable;
+
+class ProjectNotFoundException extends Exception
+{
+    public const MESSAGE = 'No s\'ha trobat cap projecte associat a aquest estudiant amb ID: %s';
+
+    public function __construct($studentId, $code = 0, Throwable $previous = null)
```
Comment (toniGitH) [inline app/Exceptions/ProjectNotFoundException.php:12]: Following Jordi's recommendation in other custom exception files, I suggest replacing ``$code = 0`` in the constructor with ``$code = 404``.

This ``$code = 404`` will be used later in ``StudentProjectsDetailController.php`` to replace the respective hard-coded 404 in each corresponding catch.

**This change has already been applied to the rest of the custom exceptions that are already merged into develop**.

File: app/Http/Controllers/api/StudentProjectsDetailController.php [hunk 3]
```diff
@@ -1,35 +1,39 @@
...
+    }
 
-        return response()->json($projects_detail);
+    public function __invoke($uuid): JsonResponse
+    {
+        try {
+            $service = $this->studentProjectsDetailService->execute($uuid);
+            return response()->json(['projects' => $service]);
+        } catch (StudentNotFoundException $e) {
+            return response()->json(['message' => $e->getMessage()], 404);
```
Comment (toniGitH) [inline app/Http/Controllers/api/StudentProjectsDetailController.php:30]: According to the previous comment I've made in ProjectNotFoundException file, the hard-coded 404 code could be replaced with ``$e->getCode()``.

And so with the rest of the exceptions.

File: app/Service/Resume/StudentProjectsDetailService.php
Comment (toniGitH) [inline app/Service/Resume/StudentProjectsDetailService.php:1]: I like the decision to extract the service sub-processes into separate functions. :+1:

Perhaps later on, they could be extracted to be used by other services? :thinking:

**Review by toniGitH — APPROVED** [review]
In my humble opinion, I think you've done a great job on this PR!!

The endpoint works correctly in Swagger locally, it's well documented, and it covers several exception scenarios.

The only thing I miss is an **acceptance test of the service**, beyond the unit test you've done, which verifies that the class can be instantiated.

If you decide to do it eventually, I recommend using ``expectException()``.

In any case, I approve this PR.

👏👏👏

**Review by FranEnLaNube — APPROVED** [review]
Well done! I've checked swagger, works fine and the tests all passed. 
I agree with the coments from @ToniGitH, the king! 🙌🤗

File: app/Exceptions/ProjectNotFoundException.php (hunk 1 above)
Comment (coxen34) [reply app/Exceptions/ProjectNotFoundException.php:2]: :dizzy_face:

File: app/Http/Controllers/api/StudentProjectsDetailController.php (hunk 3 above)
Comment (coxen34) [reply app/Http/Controllers/api/StudentProjectsDetailController.php:30]: Thanks 
:+1:
:thumbsup:

File: app/Http/Controllers/api/StudentProjectsDetailController.php [hunk 4]
```diff
@@ -1,35 +1,39 @@
...
+        $this->studentProjectsDetailService = $studentProjectsDetailService;
+    }
 
-        return response()->json($projects_detail);
+    public function __invoke($uuid): JsonResponse
+    {
+        try {
+            $service = $this->studentProjectsDetailService->execute($uuid);
+            return response()->json(['projects' => $service]);
+        } catch (StudentNotFoundException $e) {
```
[the remark this answers is not in the document]
Comment (coxen34) [reply app/Http/Controllers/api/StudentProjectsDetailController.php:29]: More legible yes!
Thanks :+1:
:thumbsup:

File: app/Service/Resume/StudentProjectsDetailService.php
Comment (coxen34) [reply app/Service/Resume/StudentProjectsDetailService.php:1]: Under investigation :monocle_face:

**Comment by coxen34** [comment]
> In my humble opinion, I think you've done a great job on this PR!!
> 
> The endpoint works correctly in Swagger locally, it's well documented, and it covers several exception scenarios.
> 
> The only thing I miss is an **acceptance test of the service**, beyond the unit test you've done, which verifies that the class can be instantiated.
> 
> If you decide to do it eventually, I recommend using `expectException()`.
> 
> In any case, I approve this PR.
> 
> 👏👏👏


@toniGitH I would like to talk about it to fully understand it.

File: storage/api-docs/api-docs.json
[the remark this answers is not in the document]
Comment (coxen34) [reply storage/api-docs/api-docs.json:1]: I will take note!!

**Pull request merged**